src/xc_integrator/replicated/device/shellbatched_replicated_xc_device_integrator_exc_vxc.hpp

```diff
@@ -119,29 +176,51 @@ void ShellBatchedReplicatedXCDeviceIntegrator<ValueType>::
   GAUXC_GENERIC_EXCEPTION("GKS NOT YET IMPLEMENTED FOR DEVICE");
 }
 
-
 template <typename ValueType>
 void ShellBatchedReplicatedXCDeviceIntegrator<ValueType>::
-  exc_vxc_local_work_( const basis_type& basis, const value_type* P, int64_t ldp, 
-                       value_type* VXC, int64_t ldvxc, value_type* EXC, value_type *N_EL,
-                       host_task_iterator task_begin, host_task_iterator task_end,
+  exc_vxc_local_work_( const basis_type& basis, const value_type* Ps, int64_t ldps,
+                            const value_type* Pz, int64_t ldpz,
+                            const value_type* Py, int64_t ldpy,
+                            const value_type* Px, int64_t ldpx,
+                            value_type* VXCs, int64_t ldvxcs,
+                            value_type* VXCz, int64_t ldvxcz,
+                            value_type* VXCy, int64_t ldvxcy,
+                            value_type* VXCx, int64_t ldvxcx, value_type* EXC, value_type *N_EL,
+                            host_task_iterator task_begin, host_task_iterator task_end,
                        incore_integrator_type& incore_integrator, XCDeviceData& device_data ) {
-
-
-  //incore_integrator.exc_vxc_local_work( basis, P, ldp, VXC, ldvxc, EXC, N_EL, task_begin, task_end, device_data );
-  //return;
-
+  const bool is_gks = (Pz != nullptr) and (Py != nullptr) and (Px != nullptr);
+  const bool is_uks = (Pz != nullptr) and (Py == nullptr) and (Px == nullptr);
+  const bool is_rks = not is_uks and not is_gks;
+  if (not is_rks and not is_uks and not is_gks) {
+    GAUXC_GENERIC_EXCEPTION("MUST BE EITHER RKS, UKS, or GKS!");
+  }	
 
   const auto nbf = basis.nbf();
   const uint32_t nbf_threshold = 8000;
   const auto& mol = this->load_balancer_->molecule();
   // Zero out integrands on host
+
   this->timer_.time_op("XCIntegrator.ZeroHost", [&](){
     *EXC  = 0.;
     *N_EL = 0.;
     for( auto j = 0; j < nbf; ++j )
     for( auto i = 0; i < nbf; ++i )
-      VXC[i + j*ldvxc] = 0.;
+      VXCs[i + j*ldvxcs] = 0.;
+
+		if (is_uks or is_gks)
+    for( auto j = 0; j < nbf; ++j )
+    for( auto i = 0; i < nbf; ++i )
+      VXCz[i + j*ldvxcz] = 0.;
+		
+		if (is_gks) {
```

Comment:
I have no idea why the indenting is so bad here, it's not like this on my machine!

Reply:
Tabs vs spaces is my guess - tell `vim` (assuming that's what you're using) to expand tabs